src/raftstore/store/snap.rs

```diff
@@ -61,223 +117,517 @@ impl Display for SnapKey {
     }
 }
 
-/// Name prefix for the self-generated snapshot file.
-const SNAP_GEN_PREFIX: &'static str = "gen";
-/// Name prefix for the received snapshot file.
-const SNAP_REV_PREFIX: &'static str = "rev";
+pub trait SendSnapshotFileBuilder {
+    fn build(&mut self, snap: &DbSnapshot, region: &Region) -> raft::Result<()>;
```

Comment:
IMO, the build for the SendSnapshotFileBuilder should return a SendSnapshotFile object, then we will use the file object later.